crosshair/register_contract_test.py

```diff
@@ -22,12 +26,110 @@ def test_register_malformed_contract():
 
 def test_register_malformed_signature():
     with pytest.raises(ContractRegistrationError):
-        # The signature is missing the parameter `self`
+        # The signature has wrong arg names (should be `a` and `b`).
         sig = Signature(
             [
-                Parameter("a", Parameter.POSITIONAL_OR_KEYWORD, annotation=int),
-                Parameter("b", Parameter.POSITIONAL_OR_KEYWORD, annotation=int),
+                Parameter("self", Parameter.POSITIONAL_OR_KEYWORD),
+                Parameter("low", Parameter.POSITIONAL_OR_KEYWORD, annotation=int),
+                Parameter("high", Parameter.POSITIONAL_OR_KEYWORD, annotation=int),
             ],
             return_annotation=int,
         )
         register_contract(Random.randint, sig=sig)
+
+
+def test_register_randint():
+    def f(x: int) -> int:
+        """
+        pre: x <= 10
+        post: _ >= x
+        """
+        return randint(x, 10)
+
+    register_contract(
+        Random.randint,
+        pre=lambda a, b: a <= b,
+        post=lambda __return__, a, b: a <= __return__ and __return__ <= b,
```

Comment:
BTW, Python will let you chain comparison operators, so `a <= __return__ <= b` should be equivalent.

Reply:
Great, thank you for letting me know! I'll update it.